feat(flat-table): add new component <Sort /> - FE-2482 (#2866)
Sage/carbon: src/components/flat-table/flat-table.stories.mdx

  </Story>
</Preview>

### With the transparent effect

> if you want to create a transparent effect to header you have to override header background styling

Reviewer: ```suggestion
To create a transparent effect for the header, override the header background styling
```

> to match the same color as a background

Reviewer: ```suggestion
to match the same color as the background
```

  </Story>
</Preview>

### With transparent effect
To create a transparent effect for the header, override the header background styling
to match the same color as the background

<Preview>
  <Story name="with transparent effect" parameters={{ info: { disable: true }}} >
    <div style={{background: '#FF785B', padding: '10px'}}>

>       <FlatTable styleOverride={{headerBackground: '#FF785B'}}>

Reviewer: See above comment about `styleOverride`s

Author: fixed in https://github.com/Sage/carbon/pull/2866/commits/840d7cd4461c614101f121bf05a818cb440074d9
